src/crypto/sha/sha256.cpp

```diff
@@ -86,4 +72,13 @@ namespace libp2p::crypto {
   HashType Sha256::hashType() const {
     return HashType::SHA256;
   }
+
+  outcome::result<libp2p::common::Hash256> sha256(
+      gsl::span<const uint8_t> input) {
+    Sha256 sha;
+    OUTCOME_TRY(sha.write(input));
+    libp2p::common::Hash256 out;
+    OUTCOME_TRY(sha.digestOut(out));
+    return out;
```

Comment:
It would be better to instantiate outcome::result and move into return it to avoid copy:
```suggestion
    outcome::result<libp2p::common::Hash256> result{libp2p::common::Hash256{}};
    OUTCOME_TRY(sha.digestOut(result.value()));
    return std::move(result);
```